golang/cosmos/x/vstorage/vstorage.go

```diff
@@ -8,28 +8,37 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
 	"github.com/Agoric/agoric-sdk/golang/cosmos/vm"
+	"github.com/Agoric/agoric-sdk/golang/cosmos/x/vstorage/types"
 )
 
 type vstorageHandler struct {
 	keeper Keeper
 }
 
 type vstorageMessage struct {
-	Method string `json:"method"`
-	Path   string `json:"key"` // TODO: rename JSON to "path"
-	Value  string `json:"value"`
+	Method string            `json:"method"`
+	Args   []json.RawMessage `json:"args"`
 }
 
 type vstorageStoreKey struct {
 	StoreName       string `json:"storeName"`
 	StoreSubkey     string `json:"storeSubkey"`
 	DataPrefixBytes string `json:"dataPrefixBytes"`
+	NoDataValue     string `json:"noDataValue"`
 }
 
 func NewStorageHandler(keeper Keeper) vstorageHandler {
 	return vstorageHandler{keeper: keeper}
 }
 
+func unmarshalPathFromArgs(args []json.RawMessage, path *string) error {
+	if len(args) == 0 {
+		return fmt.Errorf("missing 'path' argument")
+	}
+
+	return json.Unmarshal(args[0], path)
+}
```

Comment:
```suggestion
func unmarshalSinglePathFromArgs(args []json.RawMessage, path *string) error {
	if len(args) == 0 {
		return fmt.Errorf("missing 'path' argument")
	}
	if len(args) != 1 {
		return fmt.Errorf("extra arguments after 'path'")
	}

	return json.Unmarshal(args[0], path)
}
```

Reply:
Since it's coming from JS I was thinking of allowing extra args, but I guess we can be strict here.